tests/light/src/axosyslog_light/syslog_ng_config/statements/sources/network_source.py

```diff
@@ -65,7 +65,17 @@ def write_log(self, message, rate=None, transport=None, framed=None):
         self.io.write_messages([message], rate=rate, transport=transport, framed=framed)
 
     def write_logs(self, messages, rate=None, transport=None, framed=None):
-        self.io.write_messages(messages, rate=rate, transport=transport, framed=framed)
+        loggen_process = self.io.write_messages(messages, rate=rate, transport=transport, framed=framed)
+        loggen_process.wait()
+        return loggen_process
 
     def write_logs_with_proxy_header(self, proxy_version, src_ip, dst_ip, src_port, dst_port, messages, rate=None, transport=None, framed=None):
-        self.io.write_messages_with_proxy_header(proxy_version, src_ip, dst_ip, src_port, dst_port, messages, rate=rate, transport=transport, framed=framed)
+        logge_process = self.io.write_messages_with_proxy_header(proxy_version, src_ip, dst_ip, src_port, dst_port, messages, rate=rate, transport=transport, framed=framed)
+        logge_process.wait()
+        return logge_process
+
+    def write_logs_async(self, messages, rate=None, transport=None, framed=None):
+        return self.io.write_messages(messages, rate=rate, transport=transport, framed=framed)
+
+    def write_logs_with_proxy_header_async(self, proxy_version, src_ip, dst_ip, src_port, dst_port, messages, rate=None, transport=None, framed=None):
+        return self.io.write_messages_with_proxy_header(proxy_version, src_ip, dst_ip, src_port, dst_port, messages, rate=rate, transport=transport, framed=framed)
```

Comment:
[optional]

We still leak the fact that sending is done with a process. I think it would be better to have the normal and async functions in the IO layer, do the `wait()` calls there, and not return anything.

Reply:
Ok, thanks. I will rewrite this part.
Please check this commit: ec06f8474f92132dc596e900df2506d8f99702ef
Is this what you have mentioned?

Reply:
I think the process is no longer leaked, so I'm resolving this comment.